File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.*;
import java.util.*;

public class Main{
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuilder result = new StringBuilder();

        StringTokenizer st = new StringTokenizer(br.readLine().trim());
        
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine().trim());
        
        long[] sums = new long[N+1];  
        sums[0] = 0;
        long sum = 0;
        
        for(int i = 1; i < N + 1; i++){
            long num = Long.parseLong(st.nextToken());
            sum += num;
            sums[i] = sum;
        }
        
        for(int m = 0; m < M; m++){
            st = new StringTokenizer(br.readLine().trim());
            int i = Integer.parseInt(st.nextToken());
            int j = Integer.parseInt(st.nextToken());
            result.append(sums[j] - sums[i-1]).append('\n');
        }

        bw.write(result.toString());
		bw.flush();
		bw.close();
		br.close();
	}
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

input = sys.stdin.readline

N, M = map(int, input().strip().split())

nums = list(map(int, input().strip().split()))
sums = list()
sum = 0
sums.append(0)

for i in range(N):
    sum += nums[i]
    sums.append(sum)

result = []

for _ in range(M):
    i, j = map(int, input().strip().split())
    result.append(sums[j] - sums[i-1])

sys.stdout.write('\n'.join(map(str, result)) + '\n')